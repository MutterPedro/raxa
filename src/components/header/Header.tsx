import Login from './Login';
import Logo from './Logo';
import MobileMenu from './MobileMenu';
import NavBar from './NavBar';

export default function Header() {
  return (
    <header className="bg-primary text-secondary-color arvo-bold">
      <NavBar>
        <Logo />
        <div className="hidden lg:flex lg:gap-x-12">
          <a href="/" className="text-4xl leading-6">
            RAXA
          </a>
        </div>
        <Login />
      </NavBar>
      <MobileMenu />
    </header>
  );
}
