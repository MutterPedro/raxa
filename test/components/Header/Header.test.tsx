import { render } from '@testing-library/react';

import Header from '../../../src/components/header/Header';

describe('Header.tsx', function () {
  describe('Sanity tests', function () {
    it('should exist #sanity', function () {
      expect(Header).not.toBeFalsy();
    });

    it('should render correctly #sanity', function () {
      const tree = render(<Header />).asFragment();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Unit tests', function () {
    it('should have the login button #unit', function () {
      const result = render(<Header />);
      expect(result.getAllByText('Log in')[0]).toBeInTheDocument();
    });

    it('should have the home button #unit', function () {
      const result = render(<Header />);
      expect(result.getAllByText('RAXA')[0]).toBeInTheDocument();
    });
  });
});
