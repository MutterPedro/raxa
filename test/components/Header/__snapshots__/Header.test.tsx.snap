// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.tsx Sanity tests should render correctly #sanity 1`] = `
<header
  className="bg-primary text-secondary-color arvo-bold"
>
  <nav
    aria-label="Global"
    className="mx-auto flex items-center justify-between p-6 lg:px-8"
  >
    <div
      className="flex lg:flex-1"
    >
      <a
        className="-m-1.5 p-1.5"
        href="#"
      >
        <span
          className="sr-only"
        >
          Raxa
        </span>
        <img
          alt=""
          className="h-8 w-auto"
          src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
        />
      </a>
    </div>
    <div
      className="hidden lg:flex lg:gap-x-12"
    >
      <a
        className="text-4xl leading-6"
        href="#"
      >
        RAXA
      </a>
    </div>
    <div
      className="hidden lg:flex lg:flex-1 lg:justify-end"
    >
      <a
        className="text-sm font-semibold leading-6"
        href="#"
      >
        Log in 
        <span
          aria-hidden="true"
        >
          â†’
        </span>
      </a>
    </div>
  </nav>
  <div
    aria-modal="true"
    className="lg:hidden"
    role="dialog"
  >
    <div
      className="fixed inset-0 z-10"
    />
    <div
      className="fixed inset-y-0 right-0 z-10 w-full overflow-y-auto bg-white px-6 py-6 sm:max-w-sm sm:ring-1 sm:ring-gray-900/10"
    >
      <div
        className="flex items-center justify-between"
      >
        <a
          className="-m-1.5 p-1.5"
          href="#"
        >
          <span
            className="sr-only"
          >
            Your Company
          </span>
          <img
            alt=""
            className="h-8 w-auto"
            src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
          />
        </a>
        <button
          className="-m-2.5 rounded-md p-2.5 text-gray-700"
          type="button"
        >
          <span
            className="sr-only"
          >
            Close menu
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            strokeWidth="1.5"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        className="mt-6 flow-root"
      >
        <div
          className="-my-6 divide-y divide-gray-500/10"
        >
          <div
            className="space-y-2 py-6"
          >
            <div
              className="-mx-3"
            >
              <button
                aria-controls="disclosure-1"
                aria-expanded="false"
                className="flex w-full items-center justify-between rounded-lg py-2 pl-3 pr-3.5 text-base font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                type="button"
              >
                Product
                <svg
                  aria-hidden="true"
                  className="h-5 w-5 flex-none"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                >
                  <path
                    clipRule="evenodd"
                    d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
              <div
                className="mt-2 space-y-2"
                id="disclosure-1"
              >
                <a
                  className="block rounded-lg py-2 pl-6 pr-3 text-sm font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                  href="#"
                >
                  Analytics
                </a>
                <a
                  className="block rounded-lg py-2 pl-6 pr-3 text-sm font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                  href="#"
                >
                  Engagement
                </a>
                <a
                  className="block rounded-lg py-2 pl-6 pr-3 text-sm font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                  href="#"
                >
                  Security
                </a>
                <a
                  className="block rounded-lg py-2 pl-6 pr-3 text-sm font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                  href="#"
                >
                  Integrations
                </a>
                <a
                  className="block rounded-lg py-2 pl-6 pr-3 text-sm font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                  href="#"
                >
                  Automations
                </a>
                <a
                  className="block rounded-lg py-2 pl-6 pr-3 text-sm font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                  href="#"
                >
                  Watch demo
                </a>
                <a
                  className="block rounded-lg py-2 pl-6 pr-3 text-sm font-semibold leading-7 text-gray-900 hover:bg-gray-50"
                  href="#"
                >
                  Contact sales
                </a>
              </div>
            </div>
            <a
              className="-mx-3 block rounded-lg px-3 py-2 text-base font-semibold leading-7 text-gray-900 hover:bg-gray-50"
              href="#"
            >
              Features
            </a>
            <a
              className="-mx-3 block rounded-lg px-3 py-2 text-base font-semibold leading-7 text-gray-900 hover:bg-gray-50"
              href="#"
            >
              Marketplace
            </a>
            <a
              className="-mx-3 block rounded-lg px-3 py-2 text-base font-semibold leading-7 text-gray-900 hover:bg-gray-50"
              href="#"
            >
              Company
            </a>
          </div>
          <div
            className="py-6"
          >
            <a
              className="-mx-3 block rounded-lg px-3 py-2.5 text-base font-semibold leading-7 text-gray-900 hover:bg-gray-50"
              href="#"
            >
              Log in
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
